### PR TITLE
Update CSL

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

File: contrib/refresh_checksums.mk [hunk 1]
```diff
@@ -47,7 +47,7 @@ endef
 # if $(3) is "assert", we set BINARYBUILDER_LLVM_ASSERTS=1
 define checksum_dep
 checksum-$(1)-$(2)-$(3):
-	-$(MAKE) $(QUIET_MAKE) -C "$(JULIAHOME)/deps" $(call make_flags,$(1),$(2),$(3)) checksum-$(1)
+	-+$(MAKE) $(QUIET_MAKE) -C "$(JULIAHOME)/deps" $(call make_flags,$(1),$(2),$(3)) checksum-$(1)
```
Comment (staticfloat) [inline contrib/refresh_checksums.mk:50]: I'm not entirely sure what the `+` prefix here means; is this intended to print the command but not actually execute it?

File: contrib/refresh_checksums.mk [hunk 2]
```diff
@@ -79,25 +79,27 @@ $(foreach triplet,$(CLANG_TRIPLETS),$(eval $(call checksum_dep,llvm,$(triplet),a
...
-	-$(MAKE) $(QUIET_MAKE) -C "$(JULIAHOME)/deps" USE_BINARYBUILDER_LLVM=0 DEPS_GIT=0 BUILD_LLDB=1 BUILD_LLVM_CLANG=1 BUILD_CUSTOM_LIBCXX=1 USECLANG=1 checksum-llvm
+	-+$(MAKE) $(QUIET_MAKE) -C "$(JULIAHOME)/deps" USE_BINARYBUILDER_LLVM=0 DEPS_GIT=0 BUILD_LLDB=1 BUILD_LLVM_CLANG=1 BUILD_CUSTOM_LIBCXX=1 USECLANG=1 checksum-llvm
 all: checksum-llvm-special-src
 .PHONY: checksum-llvm-special-src
 
 # merge substring project names to avoid races
 pack-checksum-llvm-tools: | pack-checksum-llvm
+pack-checksum-llvm: | checksum-llvm-tools
 pack-checksum-csl: | pack-checksum-compilersupportlibraries
+pack-checksum-compilersupportlibraries: | checksum-csl
```
Comment (staticfloat) [inline contrib/refresh_checksums.mk:102]: I kind of feel we should either use `csl` or `compilersupportlibraries` but not both.

File: deps/Makefile [hunk 3]
```diff
@@ -183,21 +195,14 @@ compile: $(addprefix compile-, $(DEP_LIBS))
...
-distcleanall: $(addprefix distclean-, $(DEP_LIBS))
-	rm -rf $(build_prefix)
-getall: get-llvm get-libuv get-pcre get-openlibm get-dsfmt get-blastrampoline get-openblas get-lapack get-suitesparse get-unwind get-gmp get-mpfr get-patchelf get-utf8proc get-objconv get-mbedtls get-libssh2 get-nghttp2 get-curl get-libgit2 get-libwhich get-zlib get-p7zip get-csl
-
-# If we're building for MacOS, no matter what, `getall` should include `llvmunwind`
-ifeq ($(OS),Darwin)
-getall: get-llvmunwind
-endif
+install: version-check $(addprefix install-, $(DEP_LIBS))
+version-check: $(addprefix version-check-, $(DEP_LIBS_STAGED))
```
Comment (staticfloat) [inline deps/Makefile:199]: glad you split this out; this always confused me that it was added in `uninstallers`

**Review by staticfloat — APPROVED** [review]

**Comment by staticfloat** [comment]
macOS says:

```
make[1]: *** No rule to make target `version-check-llvmunwind', needed by `version-check'.  Stop.
```

File: contrib/refresh_checksums.mk (hunk 1 above)
Comment (vtjnash) [reply contrib/refresh_checksums.mk:50]: It tells make that the program executed on this line is `make`. It is often auto-detected, but seemed to be broken here

File: contrib/refresh_checksums.mk (hunk 2 above)
Comment (vtjnash) [reply contrib/refresh_checksums.mk:102]: I agree, it is awkward that this is the only target which doesn't install something of the same name

File: Makefile [hunk 4]
```diff
@@ -448,7 +448,7 @@ endif
 	echo "base/version_git.jl" > light-source-dist.tmp
 
 	# Download all stdlibs and include the tarball filenames in light-source-dist.tmp
-	@$(MAKE) -C stdlib getall NO_GIT=1
+	@$(MAKE) -C stdlib getall DEPS_GIT=0 USE_BINARYBUILDER=0
```
Comment (staticfloat) [inline Makefile:451]: `light-source` is not supposed to contain any dependency source; only Julia's source.  Specifying `USE_BINARYBUILDER=0` doesn't really make sense here.

File: Makefile [hunk 5]
```diff
@@ -472,7 +472,7 @@ source-dist:
 # Make tarball with Julia code plus all dependencies
 full-source-dist: light-source-dist.tmp
 	# Get all the dependencies downloaded
-	@$(MAKE) -C deps getall NO_GIT=1
+	@$(MAKE) -C deps getall DEPS_GIT=0 USE_BINARYBUILDER=0
```
Comment (staticfloat) [inline Makefile:475]: We specify `USE_BINARYBUILDER=1` and `USE_BINARYBUILDER=0` to generate two different kinds of `full-source-dist` tarballs.  You shouldn't set it here unconditionally.

**Review by staticfloat — CHANGES_REQUESTED** [review]

File: Makefile (hunk 5 above)
Comment (staticfloat) [reply Makefile:475]: X-ref: https://github.com/JuliaCI/julia-buildbot/blob/138f0c5181bcbf1c1a802674c44d19ab18d8db1a/master/doctest.py#L98-L132

Comment (vtjnash) [reply Makefile:475]: I think you misunderstand the target definitions here. The `make -C get` target would be what is used to grab all of the binary files needed to build on this machine. The `getall` target grabs all of the source files needed to build on any machine. The `getall` target therefore logically cannot be used with USE_BINARYBUILDER=1. I could add `$(MAKE) -f contrib/refresh_checksums.mk` here though, conditional on `USE_BINARYBUILDER=1`, if you like.

Comment (staticfloat) [reply Makefile:475]: We have customers that are using things like `make -C deps getall USE_BINARYBUILDER=1 USE_BINARYBUILDER_XXX=0` to generate a tarball that they can deploy to CI and patch with their own patches to deps and such without having to rebuild LLVM, for instance.  What would you suggest they use instead?

File: Makefile (hunk 4 above)
Comment (vtjnash) [reply Makefile:451]: Agreed, this was just future-proofing against it deciding to do something foolish later

File: Makefile (hunk 5 above)
Comment (vtjnash) [reply Makefile:475]: They should use `get` (or perhaps `extract`), which is the first stage of the build pipeline (or the second, which also supports adding custom patches after). The getall target is not part of the build pipeline, but is instead part of the uninstall / cleanall set, which are a list of all possible targets, not just those that currently need to be built locally.

Comment (staticfloat) [reply Makefile:475]: okay; once this is merged let's double-check that the generated tarballs look right.

**Comment by staticfloat** [comment]
Update; this is not working as intended.  The "with binarybuilder" tarball is bundling the source versions of dependencies.  :/

**Comment by vtjnash** [comment]
So you want us to add a call to refresh_checksums?

**Comment by staticfloat** [comment]
I don't think this has anything to do with refresh_checksums; when running `make full-source-dist USE_BINARYBUILDER=1`, we should not download the source distributions of dependencies, we should download the binary builder versions of the dependencies.

**Comment by staticfloat** [comment]
Bump; this still needs to be fixed.

**Comment by vtjnash** [comment]
It was unclear what you wanted. Do you want a lite-source-dist to include binaries? Do you want a full-binarysouces-dist that had some binaries and some sources? Do you just want install dist to provide binaries?

**Comment by staticfloat** [comment]
The old behavior was for `make full-source-dist USE_BINARYBUILDER=1` to create a tarball that contained the source of Julia, and the binaries of all dependents.  If you wanted the source of Julia and the source of dependents, you had to run `make full-source-dist USE_BINARYBUILDER=0`.  I don't care about the names, I just want the ability to create both of these, so that we can continue to publish tarballs that have, for instance, the source of Julia and the binaries of everything else.  It was kind of nice that you could generate tarballs with specific source/binary configurations (via e.g. `USE_BINARYBUILDER=0 USE_BINARYBUILDER_LLVM=1`).  This is primarily useful for users that carry local patches against certain deps/Julia itself, but don't want to build LLVM every time.

**Comment by vtjnash** [comment]
It sounds like you want either `make -C deps get && make full-source-dist` or `make -f contrib/refresh_checksums.mk && make full-source-dist`?

**Comment by staticfloat** [comment]
I think you're trying to use `make -f contrib/refresh_checksums.mk` as a proxy for downloading the tarballs, but that's not the behavior we had before; that will download for _all_ platforms.  We want to only download for the current platform.

**Comment by vtjnash** [comment]
`make -C deps getall` never downloaded only for the current platform (hence "all"), which is why I still think you wanted `make -C deps get`

**Comment by staticfloat** [comment]
> `make -C deps getall` never downloaded only for the current platform (hence "all"),

I just tested this, and it's exactly what it used to do.  Perhaps it's not what was intended, but it's what happens.  Just do a `git checkout v1.6.0` and run `make -C deps getall`.

**Comment by vtjnash** [comment]
Ah, yes, checking though the version history, it seems that was a regression in v1.5 relative to when I added it in 01654af4cefcb844166deb49379815b842b556f8. Whereas, the description in the Makefile has rather cryptically said (since e85a1a5967be8a9910fe333609b75262eb23e794):
```
# additionally all targets should be listed in the getall
```
where, by observation of the list, "all targets" was not conditional

**Pull request merged**